src/rpc/RPCHelpers.cpp

```diff
@@ -686,7 +680,7 @@ traverseOwnedNodes(
             std::find(std::begin(indexes), std::end(indexes), hexMarker) ==
             std::end(indexes))
         {
-            return Status(ripple::rpcINVALID_PARAMS, "Marker not found");
+            return Status(ClioError::rpcMARKER_NOT_FOUND);
```

Comment:
so rippled just returns an empty list in this case right? If so, I'd just return `AccountCursor({beast::zero,0})`. In the handlers, if we return a 0 cursor, we don't return anything to the client

Reply:
Ah, that might be a really good way if it works. Let me try that

Reply:
The only thing is that i'm not sure we can handle `validated=false` that way.

Reply:
what is `validated=false`?

Reply:
If i use a valid marker but change one character in it i get this from rippled:

```
{
  "result": {
    "account": "raeR1nAGkMm18hHpUz1ZP6wGKLA4WCh8rX",
    "account_objects": [],
    "ledger_current_index": 23293951,
    "status": "success",
    "validated": false
  }
}
```

with your suggestion clio now gives:
```
{
  "result": {
    "account": "raeR1nAGkMm18hHpUz1ZP6wGKLA4WCh8rX",
    "account_objects": [],
    "ledger_hash": "5CF222090AFB48C88FB27FD0BB9660404C39E2BFE90B47D8D2E4C5EA7D49598C",
    "ledger_current_index": 23294159,
    "validated": true,
    "status": "success"
  },
  "warnings": [
    {
      "id": 2001,
      "message": "This is a clio server. clio only serves validated data. If you want to talk to rippled, include 'ledger_index':'current' in your request"
    },
    {
      "id": 2002,
      "message": "This server may be out of date"
    }
  ]
}
```

as you can see, rippled sets `validated` to false while `clio` does not. 
If you look at my code changes in this pr you will see that i have added a way for any rpc to set `validated` and not have it be reset to `true` which is literally hardcoded atm. :) 
So to sum up - we need to handle `validated` and set it to `false` in this case.. somehow. With your idea however i can't distinguish empty results vs. this marker that was not found situation.

Reply:
I could do something like this:

```
    if (auto status = std::get_if<RPC::Status>(&next))
        return *status;

    auto const& nextMarker = std::get<RPC::AccountCursor>(next);
    if (nextMarker.isNonZero())
        response[JS(marker)] = nextMarker.toString();
    else
        response[JS(validated)] = false;

    return response;
```

but this leads to the final paged portion being returned with `validated=false` because apparently the `zero` nextMarker is used to signify end of paged data.

Reply:
oh don't worry about that, thats an intended discrepancy between clio and rippled. clio by default returns validated data, rippled by default returns not yet validated data. If you set `"ledger_index":"validated"` in your request to rippled, this discrepancy won't exist.

Reply:
I'm only worried about it because i bet that the next bug report will be about this field differ from rippled's output..
Also even for `clio` i think it makes sense that for a bogus marker we get non-validated data (if not an error like before). But i can't see an easier solution to my custom error code approach. Maybe we can ignore it for the time being to move on with the bugs and later deal with the next bug that will be reported for this difference.

Reply:
its ok, everything coming back from clio is always validated data. It can't not be, because clio only stores validated data. The only way to get not validated data from clio is if clio forwards the request to rippled. When you send a request to rippled, you should always set `"ledger_index":"validated"` before comparing to clio's response. Or set it to a number. In your example request, you set `"ledger_current_index"` instead of `"ledger_index"`

Reply:
We have been very clear to everyone from the beginning that clio only serves validated data. That won't change. It's a known and intended discrepancy. It's bad UX to serve not yet validated data